compiler/cpp/src/generate/t_cpp_generator.cc

```diff
@@ -1699,6 +1723,24 @@ void t_cpp_generator::generate_struct_ostream_operator(std::ofstream& out, t_str
   out << "}" << endl << endl;
 }
 
+/**                                                                                                                                           
+ * Generates what() method for exceptions                                                                                                     
+ */                                                                                                                                           
+void t_cpp_generator::generate_exception_what_method(std::ofstream& out, t_struct* tstruct) {
+  out << indent();
+  generate_exception_what_method_decl(out, tstruct, true);
+  out << " {" << endl;
+
+  indent_up();
+
+  out << indent() << "std::stringstream ss;" << endl;
+  out << indent() << "ss << \"TException - service has thrown: \" << *this;" << endl;
+  out << indent() << "return ss.str().c_str();" << endl;
```

Comment:
this can cause Segmenation Fault, as ss will be destroyed on function exit but pointer to it external contents is returned...

Please do try making a unit test for that functionality